hms-app/src/actions/HackathonActions.tsx

```diff
@@ -1,31 +1,39 @@
+import { Hackathon, parseHackathon, parseHackathons } from '../common/types'
+
 const core_url = process.env.REACT_APP_CORE_URL
 
-export const getListOfHackathons = () => {
+export const getListOfHackathons = (): Promise<Hackathon[]> => {
   return fetch(`${core_url}/hackathons`, {
     method: 'GET',
     headers: {
       Accept: 'application/json',
       'Content-Type': 'application/json',
     },
   })
-    .then((data) => {
-      return data.json()
+    .then((data) => data.json())
+    .then((json) => parseHackathons(json.hackathons))
+    .catch((err) => {
+      console.log(err)
+      throw err
```

Comment:
@KBernaszuk I said we could throw here, but I don't really know if that's the right approach in react when things fail from the backend :thinking: